### PR TITLE
Get rid of $projectData dependency in services

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
That way services become stateless and can be used in long-running processes

Ping @rosen-vladimirov @TsvetanMilanov

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: declarations.d.ts [hunk 1]
```diff
@@ -478,6 +478,10 @@ interface ISpawnResult {
 
 interface ISpawnFromEventOptions {
 	throwError: boolean;
+	emitOptions?: {
+		shouldEmit: boolean;
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: what's the case when I pass emitOptions, but I do not want to emit?

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request merged**